src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -910,9 +916,12 @@ export class DashboardAppController {
       if (outputSubscription) {
         outputSubscription.unsubscribe();
       }
-      if (dashboardContainer) {
-        dashboardContainer.destroy();
-      }
+      // Temp hack - there is an issue with destroying, you'll get an error if you call `updateInput` on a
+      // destroyed embeddable. Maybe we need to instead call this "unmount" or something to differentiate
+      // cleaning up the rendering from destroying, when an embeddable may be in memory but not on screen.
+      // if (dashboardContainer) {
```

Comment:
This was my first attempt as well, but the container is destroyed as soon as we navigate away from dashboard to visualize. So when visualize editor is adding an embeddable, it is already adding it to a destroyed container. That's why I thought that just passing the config around and letting the dashboard render the embeddable in the latest container is the better approach.

Reply:
I thought of another possibility this morning - setting/getting the `getLastLoadedDashboardAppDashboardInput` instead of the Embeddable instance. Now I can put back the destroy call. This also would probably be easier to preserve in session storage.  Once dashboard is in a single plugin, I'd even get rid of exposing the external setter.  Check out the latest commits.

The problem with using config is that it's not serializable, but explicit input gets stored in the URL (or should).  I haven't had a chance to play around with your PR yet, but I suspect there would be issues with that.  This also keeps the config object private for whoever is registering the embeddable factory and embeddable.  An embeddable doesn't need to use any config at all, it's an implementation detail of visualize embeddable specifically. For example, HelloWorldEmbeddable has no config object:

```
  public async create(initialInput: EmbeddableInput, parent?: IContainer) {
    return new HelloWorldEmbeddable(initialInput, parent);
  }
```

Reply:
Oh, I love the solution with exposing `dashboardInput`. I tried your solution and it works well. I'll try to make it work with visualize embeddable.